### ISSUE TITLE
Integrate and evaluate my GNN models using the existing training pipeline

### ISSUE THREAD

**Issue by Alimbj** [TryDotAtwo/MultiGPUBeamSearch#5]
## Goal

I would like to integrate one or both model variants from my repository into this project and evaluate them using the existing training and evaluation pipeline used by the models already implemented in this repository.

The main goal is to compare my architecture with the existing models under the same experimental conditions, and to compare the two alternative Level-1 encoders within my own architecture.

## My repository

https://github.com/Alimbj/Two_level_GNN_for_pancacke_graph

The relevant code is currently implemented primarily in a Jupyter notebook.

The notebook contains two alternative implementations of the Level-1 encoder. Please inspect the repository and determine how these implementations can be integrated into the existing framework.

## My model architecture

My repository contains a two-level architecture for solving combinatorial problems such as pancake sorting and potentially Rubik's Cube.

The architecture operates at two levels.

### Level 1 — element-level representation

Within each state, individual elements are represented as nodes, and their relationships to specific positions are represented by edges or other structural relations.

I currently have two alternative implementations of this first-level encoder:

1. GNN-based encoder
2. Transformer-based encoder

These are the two model variants that I want to compare.

### Level 2 — state-level representation

After the first-level processing, the element-level representations are aggregated into a representation of the complete state.

At the second level, states are represented as nodes, while valid transitions between neighboring states are represented as edges.

The second-level graph processing is shared between both variants.

Therefore, the two variants are conceptually:

```
Level-1 GNN → aggregation → Level-2 GNN
```

versus

```
Level-1 Transformer → aggregation → Level-2 GNN
```

The Level-2 architecture and downstream components should remain the same for both variants wherever possible.

The two Level-1 encoders should therefore be treated as alternative implementations of the same architectural component, rather than as two completely independent models.

## Main requirement

The comparison should be as fair and consistent as possible.

Where applicable, my models should use the same:

* dataset and problem instances;
* data split;
* training procedure;
* optimization setup;
* number of training steps/epochs;
* evaluation procedure;
* evaluation metrics;
* relevant preprocessing;
* hardware and batching conditions

as the existing models in this repository.

The existing training and evaluation protocol should be treated as the baseline experimental setup.

Do not change the existing models or their experimental protocol simply to make my models perform better.

If model-specific adjustments are technically necessary for valid training, document them explicitly.

## Important — inspect both codebases first

Before implementing anything, please inspect:

1. The existing models in this repository.
2. The existing training pipeline.
3. The dataset generation/loading pipeline.
4. The existing evaluation procedure.
5. The expected model interface.
6. My repository and both Level-1 encoder implementations.
7. The way my Level-1 representations are aggregated and passed to Level 2.

First determine how the two codebases differ and what needs to be adapted for integration.

Do not begin a major refactoring or redesign before this analysis is complete.

If some information, files, checkpoints, dependencies, or assumptions are missing, report them before making major changes.

Please inspect my actual implementation to determine the exact representation of nodes, edges, positions, and the aggregation mechanism rather than inferring or redesigning these components from this high-level description.

## Architectural constraints

The two Level-1 encoders should remain alternative implementations of the same component.

The purpose of the experiment is to compare the Level-1 GNN and Transformer representations while keeping the Level-2 architecture and the rest of the experimental setup as consistent as possible.

Please do not redesign the Level-2 architecture unless this is required for compatibility with the existing framework.

Preserve the original architecture and logic of my models as much as possible.

Adapt the interface between my models and the existing framework rather than redesigning the models themselves.

Do not silently change the graph representation or model semantics simply to make the integration easier. If such a change is necessary, document it explicitly.

## Implementation

After understanding both codebases, integrate the selected model variant(s) into the existing framework with minimal changes to the existing code.

Prefer adapting my models to the existing framework rather than creating a completely separate training pipeline.

Reuse the existing dataset, training, and evaluation infrastructure whenever it is compatible with my architecture.

If both Level-1 variants can be integrated without substantial changes, integrate and evaluate both.

If only one can be integrated without changing the intended architecture or experimental setup, explain the limitation and identify the required changes for the other variant.

## Training and evaluation

Train and evaluate my model variant(s) using the existing training and evaluation procedure used for the baseline models.

The objective is to obtain results that allow a direct and meaningful comparison with the existing models.

Please clearly document:

* which of my model variants were used;
* how each model was integrated;
* what adaptations were necessary;
* which dataset and problem instances were used;
* what training configuration was used;
* what evaluation configuration was used;
* the resulting metrics;
* any differences between my implementation and the existing models that could affect the comparison.

If an exact apples-to-apples comparison is not possible, do not hide or normalize the difference. Clearly identify the specific difference and explain its potential effect on the comparison.

## Constraints

* Preserve the original architecture and logic of my models as much as possible.
* Adapt the model interface rather than redesigning the models.
* Do not rewrite the existing project unnecessarily.
* Do not modify existing baseline models or their experimental protocol unless this is necessary for compatibility.
* Do not introduce a separate experimental pipeline when the existing pipeline can be reused.
* Do not make undocumented architectural or training changes solely to improve performance.

The first step should be to inspect and understand both repositories and report any compatibility issues or missing information before making substantial changes.
